app/presenters/result_set_presenter.rb

```diff
@@ -49,8 +64,91 @@ def selected_filter_descriptions
     }.reject(&:empty?)
   end
 
+  def documents_by_facets
+    return [] unless grouped_display?
+
+    sector_facets = {}
+    all_businesses = { all_businesses: empty_facet_group("all_businesses", "All Businesses") }
+    other_facets = {}
+
+    primary_facet = :sector_business_area
+    primary_facet_group = %W(sector_business_area business_activity)
+
+    facet_filters = @filter_params.without("order", "keywords")
+
+    documents.select! { |d| d[:document][:metadata].present? }
+    sorted_documents = documents.sort { |x, y| x[:document][:title] <=> y[:document][:title] }
+    sorted_documents = sorted_documents.sort do |x, y|
+      return -1 if x[:document][:promoted]
```

Comment:
It would be good if we could somehow have a common method for this and `sort_by_promoted`, so if we ever changed the promoted sorted logic we wouldn't have to update it in 2 places.

Reply:
Agreed, I think this is possible.

Reply:
https://github.com/alphagov/finder-frontend/pull/856/commits/125b3dde5fd533d3dc1287e01923fe91b734a20b tidies this up a bit